File: go/leap/leap.go
package leap

// IsLeapYear determines whether a given year is a leap year or not.
func IsLeapYear(year int) bool {
    // Return true if the year is divisible by 4 and
    // either divisible by 400
    // or not divisible by 100.
    // Return false otherwise.
	return year%4 == 0 && (year%100 != 0 || year%400 == 0)
}

File: erlang/leap/leap.erl
-module(leap).
-export([leap_year/1]).

%% @doc Determine whether a given year is a leap year.
-spec leap_year(integer()) -> boolean().
leap_year(Y) when is_integer(Y) ->
	  case Y rem 4 of
		    0 -> Y rem 100 =/= 0 orelse Y rem 400 =:= 0;
		    _ -> false
	  end.

